tuf/api/metadata.py

```diff
@@ -512,14 +606,64 @@ def update(
             self, rolename: str, version: int, length: Optional[int] = None,
             hashes: Optional[JsonDict] = None) -> None:
         """Assigns passed (delegated) targets role info to meta dict. """
+
+        # TODO: Consider renaming this function:
+        # see: https://github.com/theupdateframework/tuf/issues/1230
         metadata_fn = f'{rolename}.json'
 
-        self.meta[metadata_fn] = {'version': version}
-        if length is not None:
-            self.meta[metadata_fn]['length'] = length
+        self.meta[metadata_fn] = MetadataInfo(version, length, hashes)
+
+
+class TargetInfo:
+    """A container with information about a particular target file.
+    Instances of TargetInfo are used as values in a dictionary
+    called "targets" in Targets.
+
+    Attributes:
+        length: An integer indicating the length of the target file.
+        hashes: A dictionary containing hash algorithms and the
+            hashes resulted from applying them over the target file::
```

Comment:
Similar to the above suggestion, I'd welcome feedback on whether this is clearer or not?
```suggestion
        hashes: A dictionary mapping hash algorithms to the
            hashes resulting from applying them over the metadata file
            contents::
```